### ISSUE TITLE
Add code examples for `boa_interner`

### ISSUE THREAD

**Issue by jedel1043** [boa-dev/boa#2337]
The `boa_interner` crate has an `allow(rustdoc::missing_doc_code_examples)` on its root. Ideally, this crate should have some usage snippets on every `pub` function.

**Comment by Joymfl** [comment]
Can I pick this up?

**Comment by jedel1043** [comment]
> Can I pick this up?

Assigned!

**Comment by Joymfl** [comment]
Hey, currently I'm looking at how the `pub` functions are being used in the `boa_interner/src/tests.rs` and then documenting code examples for the same, but I see more functions that aren't being used in the tests. Any tips on how to understand and document them?

**Comment by Joymfl** [comment]
Sounds good 👍🏾 . I'll try that and get back to you

**Comment by postmeback** [comment]
@Joymfl , any update on this ?

**Comment by Razican** [comment]
I don't think this issue should be closed, since the related PR added a couple of tests, but no examples in the documentation of the interner.

**Comment by postmeback** [comment]
> I don't think this issue should be closed, since the related PR added a couple of tests, but no examples in the documentation of the interner.

Can you give me an reference, where should I add the documentation ?

**Comment by postmeback** [comment]
Hi, Can i get any assistance on the above comment, please?

**Comment by Razican** [comment]
Hi @postmeback, sorry for not writing here!

I was checking the code, and in fact, the mentioned lint is unstable (https://github.com/rust-lang/rust/issues/101730). You could in theory enable it by adding `#![feature(rustdoc_missing_doc_code_examples)]` and `#![warn(rustdoc::missing_doc_code_examples)]` in `lib.rs` in `boa_interner`, and then run `cargo +nightly doc` to get the relevant warnings.

This can help you get the warnings on where some usage examples need to be added in the API. `boa_interner` was selected for this since its public API is small enough.

For each warning, the idea is to add some code examples in the doc comments, that show how to use it.

**Comment by postmeback** [comment]
I followed your steps, I did not get any warnings in the `boa_interner` folder.

![image](https://github.com/boa-dev/boa/assets/14862687/b2442c17-4c10-402d-a16d-cf871c8d0737)

Requesting you to provide your feedback on this. :)

**Comment by jedel1043** [comment]
> I followed your steps, I did not get any warnings in the `boa_interner` folder.

Saw that you ran `cargo +nightly fmt`. You would need to run `cargo +nightly doc` instead to see the warnings.

**Comment by Paul-Folbrecht** [comment]
I see this is still open - I'm willing to grab it.

**Comment by KaustubhOG** [comment]
Hi! @jedel1043 I'd like to work on this issue. I'll add the missing doc code examples to boa_interner. Looking forward to contributing!
